### PR TITLE
RSCBC-50: Implement transcoding

### PR DESCRIPTION
**Pull request by chvck**

### CODE REVIEW

File: sdk/couchbase/src/results/get_result.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+use crate::error;
+use crate::transcoder::{DefaultTranscoder, Transcoder};
+use bytes::Bytes;
+use serde::de::DeserializeOwned;
+
+pub struct GetResult {
```
Comment (Matt-Woz) [inline sdk/couchbase/src/results/get_result.rs:6]: Are we intentionally omitting CAS for now (and I suppose expiry too)? If so should we create a ticket/comment to get back to it?

Comment (chvck) [reply sdk/couchbase/src/results/get_result.rs:6]: Yeah this change is just transcoding specific. There's already an issue for doing kv ops so it'll get covered in that.

File: sdk/couchbase/src/transcoder_binary.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+use crate::error;
+use crate::transcoder::{decode_common_flags, CompressionType, DataType, Transcoder};
+use bytes::Bytes;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use serde_binary::binary_stream;
+
+pub struct BinaryTranscoder {}
```
[the remark this answers is not in the document]
Comment (chvck) [reply sdk/couchbase/src/transcoder_binary.rs:8]: Hmm I don't think so, possibly we should maybe delete it though. It very much doesn't follow the rules of RawBinaryTranscoder, and isn't really that. This takes any data and turns it into binary data, RawBinaryTranscoders takes bytes and just does a passthrough.

[the remark this answers is not in the document]
Comment (Matt-Woz) [reply sdk/couchbase/src/transcoder_binary.rs:8]: Ah ok, fair enough

**Review by Matt-Woz — APPROVED** [review]

**Pull request merged**